tests/components/tellduslive/test_config_flow.py

```diff
@@ -0,0 +1,230 @@
+    assert result['reason'] == 'already_setup'
+
+    with patch.object(hass.config_entries, 'async_entries', return_value=[{}]):
+        result = await flow.async_step_import(None)
+    assert result['type'] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result['reason'] == 'already_setup'
+
+
+async def test_full_flow_implementation(hass, mock_tellduslive):
+    """Test registering an implementation and finishing flow works."""
+    flow = init_config_flow(hass)
+    result = await flow.async_step_discovery(['localhost', 'tellstick'])
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['step_id'] == 'user'
+
+    result = await flow.async_step_user()
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['step_id'] == 'user'
+
+    result = await flow.async_step_user({'host': 'localhost'})
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['step_id'] == 'auth'
+    assert result['description_placeholders'] == {
+        'auth_url': 'https://example.com',
+        'app_name': APPLICATION_NAME,
+    }
+
+    result = await flow.async_step_auth('')
+    assert result['type'] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result['title'] == 'localhost'
+    assert result['data']['host'] == 'localhost'
+    assert result['data']['scan_interval'] == 60
+    assert result['data']['session'] == {'token': 'token', 'host': 'localhost'}
+
+
+async def test_step_import(hass, mock_tellduslive):
+    """Test that we trigger auth when configuring from import."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_import({
+        KEY_HOST: DOMAIN,
+        KEY_SCAN_INTERVAL: 0,
+    })
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['step_id'] == 'auth'
+
+
+async def test_step_import_add_host(hass, mock_tellduslive):
+    """Test that we add host and trigger user when configuring from import."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_import({
+        KEY_HOST: 'localhost',
+        KEY_SCAN_INTERVAL: 0,
+    })
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['step_id'] == 'user'
+
+
+async def test_step_import_no_config_file(hass, mock_tellduslive):
+    """Test that we trigger user with no config_file configuring from import."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_import({ KEY_HOST: 'localhost', KEY_SCAN_INTERVAL: 0, })
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['step_id'] == 'user'
+
+
+async def test_step_import_load_json_matching_host(hass, mock_tellduslive):
+    """Test that we add host and trigger user when configuring from import."""
+    flow = init_config_flow(hass)
+
+    with patch('homeassistant.components.tellduslive.config_flow.load_json', return_value={'tellduslive': {}}):
+        with patch('os.path.isfile'):
```

Comment:
It's enough with a single `with...:` statement for both patches. Separate them with comma.